### ISSUE TITLE
caseOrder: error: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by saromanov** [go-critic/go-critic#779]
Hi! I found, that in the case if type on the switch statement doesn't exist, go-critic returns panic
### Example
```go
type test struct {
	value interface{}
}

func do(t *test) {
	switch v := t.value.(type) {
	case MyType: //MyType is not defined
		fmt.Println("there is not exist")
	case string:
		fmt.Println("value is string")
	default:
		fmt.Println()
	}
}
```
### Output
```sh
gocritic check-project -withExperimental .
caseOrder: error: runtime error: invalid memory address or nil pointer dereference
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x790878]

goroutine 95 [running]:
github.com/go-critic/go-critic/cmd/criticize.(*linter).checkFile.func1.1(0xc0003a2130, 0xc000381a40)
	/home/test/gowork/src/github.com/go-critic/go-critic/cmd/criticize/main.go:311 +0x138
panic(0x853d20, 0xc4cb20)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/go-critic/go-critic/lint.(*caseOrderChecker).checkTypeSwitch(0xc000142290, 0xc00012dbc0)
	/home/test/gowork/src/github.com/go-critic/go-critic/lint/caseOrder_checker.go:59 +0x138
github.com/go-critic/go-critic/lint.(*caseOrderChecker).VisitStmt(0xc000142290, 0x954a40, 0xc00012dbc0)
	/home/test/gowork/src/github.com/go-critic/go-critic/lint/caseOrder_checker.go:37 +0x76
github.com/go-critic/go-critic/lint/internal/astwalk.(*stmtWalker).WalkFile.func1(0x9516e0, 0xc00012dbc0, 0x950a20)
	/home/test/gowork/src/github.com/go-critic/go-critic/lint/internal/astwalk/stmt_walker.go:19 +0x90
go/ast.inspector.Visit(0xc000035cd0, 0x9516e0, 0xc00012dbc0, 0xc0001c9540, 0xc0001c9540)
	/usr/local/go/src/go/ast/walk.go:373 +0x3a
go/ast.Walk(0x94f620, 0xc000035cd0, 0x9516e0, 0xc00012dbc0)
	/usr/local/go/src/go/ast/walk.go:52 +0x66
go/ast.walkStmtList(0x94f620, 0xc000035cd0, 0xc00012aa00, 0x1, 0x1)
	/usr/local/go/src/go/ast/walk.go:32 +0x9e
go/ast.Walk(0x94f620, 0xc000035cd0, 0x950ba0, 0xc00012dbf0)
	/usr/local/go/src/go/ast/walk.go:224 +0x1b67
go/ast.Inspect(0x950ba0, 0xc00012dbf0, 0xc000035cd0)
	/usr/local/go/src/go/ast/walk.go:385 +0x4b
github.com/go-critic/go-critic/lint/internal/astwalk.(*stmtWalker).WalkFile(0xc00012ab30, 0xc000160f00)
	/home/test/gowork/src/github.com/go-critic/go-critic/lint/internal/astwalk/stmt_walker.go:17 +0x10b
github.com/go-critic/go-critic/lint.(*Checker).Check(0xc000381a40, 0xc000160f00, 0xc0003a2130, 0xc000381a40, 0xc0001c7650)
	/home/test/gowork/src/github.com/go-critic/go-critic/lint/lint.go:99 +0x46
github.com/go-critic/go-critic/cmd/criticize.(*linter).checkFile.func1(0xc0003a2130, 0xc000160f00, 0xc0000cc090, 0xc000381a40)
	/home/test/gowork/src/github.com/go-critic/go-critic/cmd/criticize/main.go:319 +0x87
created by github.com/go-critic/go-critic/cmd/criticize.(*linter).checkFile
	/home/test/gowork/src/github.com/go-critic/go-critic/cmd/criticize/main.go:301 +0xac
lint error: exit status 2
```
When i import package with `MyType` definition, its executing fine

**Comment by Quasilyte** [comment]
Nice!
Thank you for reporting it.
We'll fix that.

**Comment by fexolm** [comment]
Should be fixed by #808.

**Comment by MartinYangTW** [comment]
The same error message in the following versions
$rke up --update-only --config=config.yml
rke version v0.2.8

INFO[0035] [certificates] Deploying kubernetes certificates to Cluster nodes
INFO[0040] [reconcile] Rebuilding and updating local kube config
INFO[0040] Successfully Deployed local admin kubeconfig at [./kube_config_config.yml]
INFO[0040] [certificates] Successfully deployed kubernetes certificates to Cluster nodes
INFO[0040] [reconcile] Reconciling cluster state
INFO[0040] [reconcile] Check etcd hosts to be deleted
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x124261d]

goroutine 1 [running]:
github.com/rancher/rke/vendor/github.com/urfave/cli.HandleAction.func1(0xc000588d68)
        /go/src/github.com/rancher/rke/vendor/github.com/urfave/cli/app.go:472 +0x287
panic(0x13f7860, 0x2572030)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/rancher/rke/vendor/k8s.io/client-go/util/cert.EncodeCertPEM(...)
        /go/src/github.com/rancher/rke/vendor/k8s.io/client-go/util/cert/pem.go:69
github.com/rancher/rke/cluster.reconcileEtcd(0x18cf5a0, 0xc000040080, 0xc000281200, 0xc000280900, 0xc0000fe960, 0x0, 0x0)
        /go/src/github.com/rancher/rke/cluster/reconcile.go:175 +0x13d
github.com/rancher/rke/cluster.ReconcileCluster(0x18cf5a0, 0xc000040080, 0xc000280900, 0xc000281200, 0x0, 0x0, 0x7ffda6002cff, 0xa, 0x0, 0x0, ...)
        /go/src/github.com/rancher/rke/cluster/reconcile.go:43 +0x22a
github.com/rancher/rke/cmd.ClusterUp(0x18cf5a0, 0xc000040080, 0x0, 0x0, 0x0, 0x0, 0x0, 0x7ffda6002cff, 0xa, 0x0, ...)
        /go/src/github.com/rancher/rke/cmd/up.go:133 +0x650
github.com/rancher/rke/cmd.clusterUpFromCli(0xc0000e9540, 0x0, 0x0)
        /go/src/github.com/rancher/rke/cmd/up.go:242 +0x60e
reflect.Value.call(0x138d4a0, 0x1671228, 0x13, 0x15bf142, 0x4, 0xc0004cad08, 0x1, 0x1, 0xc000240140, 0x412324, ...)
        /usr/local/go/src/reflect/value.go:447 +0x461
reflect.Value.Call(0x138d4a0, 0x1671228, 0x13, 0xc0004cad08, 0x1, 0x1, 0xc00005f418, 0x140, 0xc00005f3f0)
        /usr/local/go/src/reflect/value.go:308 +0xa4
github.com/rancher/rke/vendor/github.com/urfave/cli.HandleAction(0x138d4a0, 0x1671228, 0xc0000e9540, 0x0, 0x0)
        /go/src/github.com/rancher/rke/vendor/github.com/urfave/cli/app.go:481 +0x1ff
github.com/rancher/rke/vendor/github.com/urfave/cli.Command.Run(0x15beba4, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x15d08a1, 0x14, 0x0, ...)
        /go/src/github.com/rancher/rke/vendor/github.com/urfave/cli/command.go:186 +0x8d1
github.com/rancher/rke/vendor/github.com/urfave/cli.(*App).Run(0xc000001680, 0xc00003a080, 0x4, 0x4, 0x0, 0x0)
        /go/src/github.com/rancher/rke/vendor/github.com/urfave/cli/app.go:235 +0x601
main.mainErr(0x189ad60, 0xc00000e018)
        /go/src/github.com/rancher/rke/main.go:56 +0xf94
main.main()
        /go/src/github.com/rancher/rke/main.go:19 +0x43

**Comment by quasilyte** [comment]
@MartinYangTW, I can't see anything coming from the go-critic in the stack trace. Could you check which app is causing that, since I may be lacking some info. It seems unrelated.

**Comment by MartinYangTW** [comment]
> I can't see anything coming from the go-critic in the stack trace. Could you check which app is causing that, since I may be lacking some info. It seems unrelated.

Sorry that I missed this message, but I found this problem on v0.3.1 has been fixed.
